docs/content/guides/developer/sui-101/sign-and-send-txn.mdx

```diff
@@ -0,0 +1,249 @@
+
+    // import a keypair from a base64 encoded 32-byte `private key`.
+    let _skp_import_no_flag_0 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+    let _skp_import_no_flag_1 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+    let _skp_import_no_flag_2 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+
+    // import a keypair from a base64 encoded 33-byte `flag || private key`. The signature scheme is determined by the flag.
+    let _skp_import_with_flag_0 =
+        SuiKeyPair::decode_base64("ANRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_1 =
+        SuiKeyPair::decode_base64("AdRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_2 =
+        SuiKeyPair::decode_base64("AtRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+
+    // replace `skp_determ_0` with the variable names above
+    let pk = skp_determ_0.public();
+    let sender = SuiAddress::from(&pk);
+```
+
+Next, sign transaction data constructed using an example programmable transaction block with default gas coin, gas budget, and gas price. See [Building Programmable Transaction Blocks](./building-ptb.mdx) for more information. 
+
+```
+    // construct an example programmable transaction. 
+    let pt = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        builder.pay_sui(vec![sender], vec![1]).unwrap();
+        builder.finish()
+    };
+
+    let gas_budget = 5_000_000;
+    let gas_price = sui_client.read_api().get_reference_gas_price().await?;
+
+    // create the transaction data that will be sent to the network.
+    let tx_data = TransactionData::new_programmable(
+        sender,
+        vec![gas_coin.object_ref()],
+        pt,
+        gas_budget,
+        gas_price,
+    );
+```
+
+Commit a signature to the Blake2b hash digest of the intent message (`intent || bcs bytes of tx_data`). 
+
+```Rust
+    // derive the digest that the keypair should sign on. 
+    let intent_msg = IntentMessage::new(Intent::sui_transaction(), tx_data);
+    let raw_tx = bcs::to_bytes(&intent_msg).map_err(|_| TransactionError::ParsingError)?;
+    let mut hasher = sui_types::crypto::DefaultHash::default();
+    hasher.update(raw_tx.clone());
+    let digest = hasher.finalize().digest;
+
+    // use SuiKeyPair to sign the digest. 
+    let sui_sig = skp.sign(&digest);
+
+    // verify the signature against the transaction locally. if this fails the transaction probably will fail submitting. 
```

Comment:
```suggestion
    // verify the signature against the transaction locally. If this fails, the transaction will probably fail submitting. 
```
What do we mean by `fail submitting` - is it `fail to be submitted to be executed?` or it will simply fail and an error will be return. Let's be precise here, it is important to accurately describe it.

Reply:
im not sure i understand the difference. you can submit whatever you want, but if an invalid signature is submitted, it will return a Signature Error, which is why i provided verify signature locally function bc ppl do like figure this out locally.

Reply:
Well, we are a bit vague: if the signature is invalid, then it will return an error and thus there's no `probably will fail submitting`, right?